services/payments/tests/unit/test_services_auto_recharge_listener.py

```diff
@@ -78,3 +116,282 @@ async def test_process_event_functions(
+
+        con.execute(
+            payments_methods.insert().values(
+                **random_payment_method(
+                    payment_method_id=_primary_payment_method_id,
+                    wallet_id=wallet_id,
+                    state="SUCCESS",
+                    completed_at=_completed_at,
+                )
+            )
+        )
+        con.execute(
+            payments_autorecharge.insert().values(
+                **random_payment_autorecharge(
+                    primary_payment_method_id=_primary_payment_method_id,
+                    wallet_id=wallet_id,
+                )
+            )
+        )
+
+        yield
+
+        con.execute(payments_methods.delete())
+        con.execute(payments_autorecharge.delete())
+
+
+@pytest.fixture()
+def wallet_id(faker: Faker):
+    return faker.pyint()
+
+
+@pytest.fixture
+async def mocked_pay_with_payment_method(mocker: MockerFixture) -> mock.AsyncMock:
+    return mocker.patch(
+        "simcore_service_payments.services.payments.PaymentsGatewayApi.pay_with_payment_method",
+        return_value=AckPaymentWithPaymentMethod.construct(
+            **AckPaymentWithPaymentMethod.Config.schema_extra["example"]
+        ),
+    )
+
+
+@pytest.fixture
+async def mock_rpc_server(
+    rabbitmq_rpc_client: Callable[[str], Awaitable[RabbitMQRPCClient]],
+    mocker: MockerFixture,
+) -> RabbitMQRPCClient:
+    rpc_client = await rabbitmq_rpc_client("client")
+    rpc_server = await rabbitmq_rpc_client("mock_server")
+
+    router = RPCRouter()
+
+    # mocks the interface defined in the webserver
+
+    @router.expose()
+    async def get_credit_amount(
+        dollar_amount: ProductName, product_name: ProductName
+    ) -> CreditResultGet:
+        return CreditResultGet.parse_obj(
+            CreditResultGet.Config.schema_extra["examples"][0]
+        )
+
+    await rpc_server.register_router(router, namespace=WEBSERVER_RPC_NAMESPACE)
+
+    # mock returned client
+    mocker.patch(
+        "simcore_service_payments.services.auto_recharge_process_message.get_rabbitmq_rpc_client",
+        return_value=rpc_client,
+    )
+
+    return rpc_client
+
+
+async def _assert_payments_transactions_db_row(postgres_db) -> PaymentsTransactionsDB:
+    async for attempt in AsyncRetrying(
+        wait=wait_fixed(0.2),
+        stop=stop_after_delay(10),
+        retry=retry_if_exception_type(AssertionError),
+        reraise=True,
+    ):
+        with attempt, postgres_db.connect() as con:
+            result = con.execute(sa.select(payments_transactions))
+            row = result.first()
+            assert row
+            return PaymentsTransactionsDB.from_orm(row)
+
+
+async def test_process_message__whole_autorecharge_flow_success(
+    app: FastAPI,
+    rabbitmq_client: Callable[[str], RabbitMQClient],
+    wallet_id: int,
+    populate_test_db: None,
+    mocked_pay_with_payment_method: mock.AsyncMock,
+    mock_rpc_server: RabbitMQRPCClient,
```

Comment:
for example you have it here, you could consider checking whether it was called but since you return the client, you can't